cep-20.1.md

```diff
@@ -0,0 +1,352 @@
+  format
+
+## Major differences with conda-build
+
+- no full Jinja2 support: no block support `{% set ...` support, only string
+  interpolation. Variables can be set in the toplevel "context" which is valid
+  YAML (all new features should be native to YAML specs)
+- Jinja variable syntax is changed to begin with `${{` so that it becomes valid
+  YAML, e.g. `- ${{ version }}`
+- Selectors use a YAML dictionary with `if / then / else` (vs. comments in
+  conda-build) and are only allowed in lists (dependencies, scripts, ...). The
+  syntax looks like:
+  ```yaml
+  - if: win
+    then: this
+    else: that  # optional
+  ```
+- for inline values, the Jinja ternary operator can be used, e.g. `number: ${{ 0
+  if linux else 100 }}`
+
+## Selectors
+
+Selectors in the new spec are only allowed in lists and take an explicit `if /
+then / else` syntax.
+
+For example the following `script` section:
+
+```yaml
+script:
+  - if: unix
+    then: |
+      # this is the unix script
+  - if: win
+    then: |
+      @rem a script for batch
+```
+
+The same could have been expressed with an `else`:
+
+```yaml
+script:
+  - if: unix
+    then: |
+      # this is the unix script
+    else: |
+      @rem a script for batch
+```
+
+This is a valid YAML dictionary. Selector if statements are simple boolean
+expressions and follow Python syntax. The following selectors are all valid:
+
+```
+win and arm64
+(osx or linux) and aarch64
+something == "test"
+```
+
+If the value of a selector statement is a list, it extends the "outer" list. For
+example:
+
+```yaml
+build:
+  - ${{ compiler('cxx') }}
+  - if: unix
+    then:
+      - make
+      - cmake
+      - pkg-config
+```
+
+evaluates for `unix == true` to a list with elements `[${{ compiler('cxx') }},
+make, cmake, pkg-config]`.
+
+### The cmp function for variant selection
+
+Furthermore, we have a special "cmp" function that can be used to run a check
+against a selected variant version. The `cmp` function looks like the following:
+
+```
+cmp(python, "3.6")
+cmp(python, ">=3.6")
+cmp(python, ">=3.8,<3.10")
+etc
+```
+
+This can be used in a selector like so:
+
+```yaml
+requirements:
+  build:
+    - if: cmp(python, ">=3.6,<3.10")
+      then: dataclasses
+    # The following syntax is also valid and equivalent
+    - ${{ "dataclasses" if cmp(python, ">=3.6,<3.10") }}
+```
+
+This functionality generalizes and replaces the previous special variables such
+as `py2k`, `py3k`, `py36`, `py37`, and works just as well for NumPy, Ruby, R, or
+any other variant that might be of interest in the future.
+
+### Preprocessing selectors
+
+You can add selectors to any item, and the selector is evaluated in a
+preprocessing stage. If a selector evaluates to `true`, the item is flattened
+into the parent element. If a selector evaluates to `false`, the item is
+removed.
+
+```yaml
+source:
+  - if: not win
+    then:
+      # note that we omit the `-`, both is valid
```

Comment:
I can see this becoming a bit confusing down the line. Are there cases were we do need a nested list that should not be flattened?

Maybe we can have one more key in the `if` dictionary? e.g. `flatten: true`?

Reply:
No there is no case with nested lists allowed